components/sst_importer/src/sst_importer.rs

```diff
@@ -1958,6 +1979,49 @@ mod tests {
         assert_eq!(err.kind(), io::ErrorKind::TimedOut);
     }
 
+    #[test]
+    fn test_update_config_memory_use_ratio() {
+        // create SstImpoter with default.
+        let cfg = Config {
+            memory_use_ratio: 0.3,
+            ..Default::default()
+        };
+        let import_dir = tempfile::tempdir().unwrap();
+        let importer = SstImporter::new(&cfg, import_dir, None, ApiVersion::V1).unwrap();
+        let mem_limit_old = importer.mem_limit.load(Ordering::SeqCst);
+
+        // create new config and get the diff config.
+        let cfg_new = Config {
+            memory_use_ratio: 0.1,
+            ..Default::default()
+        };
+        let change = cfg.diff(&cfg_new);
+
+        // create config manager and update config.
+        let mut cfg_mgr = ImportConfigManager::new(cfg);
+        cfg_mgr.dispatch(change).unwrap();
+        importer.update_config_memory_use_ratio(cfg_mgr);
+
+        let mem_limit_new = importer.mem_limit.load(Ordering::SeqCst);
+        assert!(mem_limit_old > mem_limit_new);
+
+        let delt = (mem_limit_old - mem_limit_new * 3) as f64 / mem_limit_old as f64;
+        assert!(delt < 0.000001)
```

Comment:
Should there be `assert!(f64::abs(delt - 1) < f64::EPSILON)`?

Reply:
Or we may remove the `/ mem_limit_old as f64` in the `delt` expression, it seems that dividing an non-zero value generally would yield a near-to-zero result.

Reply:
The calcualte_usage_mem() gets the u64 value, and the mem_limit_old - mem_limit_new * 3 = 1, the delt could > f64:EPSILON